### PR TITLE
Feature/query processor

### PR DESCRIPTION
**Pull request by grimadas**
## What is this pull request?
Query execution and validation
   
## Why do you implement it? Why do we need this pull request?
Each query should be validated and  execution  in Ametsuchi. QueryProcessingFactory allows to execute and validate each query. For example, Torii will use QueryExecutionFactory via ThreadPool. 
  
## How to use the features provided in the pull request?
Tests on validation  and execution are provided in: 
`test/module/irohad/validation/query_execution.cpp`.

### CODE REVIEW

File: irohad/model/impl/query_execution.cpp [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+#include "model/queries/responses/transactions_response.hpp"
+
+iroha::model::QueryProcessingFactory::QueryProcessingFactory(
+    ametsuchi::WsvQuery& wsvQuery, ametsuchi::BlockQuery& blockQuery)
+    : _wsvQuery(wsvQuery), _blockQuery(blockQuery) {}
+
+bool iroha::model::QueryProcessingFactory::validate(
+    const model::GetAccount& query) {
+  auto creator = _wsvQuery.getAccount(query.creator_account_id);
+  // TODO: check signatures
```
Comment (takemiyamakoto) [inline irohad/model/impl/query_execution.cpp]: This is kind of important ^_^

File: irohad/model/queries/get_account_assets.hpp [hunk 2]
```diff
@@ -27,8 +27,9 @@ namespace iroha {
     /**
      * Query for get all account's assets and balance
      */
-    struct GetAccountAssets : Query {
+    struct GetAccountAsset : Query {
```
Comment (takemiyamakoto) [inline irohad/model/queries/get_account_assets.hpp]: GetAccountAssets seems to be better English

File: irohad/model/queries/responses/account_assets_response.hpp [hunk 3]
```diff
@@ -18,15 +18,15 @@
...
+#include "model/query_response.hpp"
 
 namespace iroha {
   namespace model {
 
-    struct AccountAssetsResponse : public QueryResponse {
-
-      std::vector<AccountAsset> assets;
+    struct AccountAssetResponse : public QueryResponse {
+      AccountAsset acc_asset;
```
Comment (takemiyamakoto) [inline irohad/model/queries/responses/account_assets_response.hpp]: acct is the typical English abbreviation of account

File: irohad/model/query_execution.hpp [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+      QueryProcessingFactory(ametsuchi::WsvQuery& wsvQuery,
+                             ametsuchi::BlockQuery& blockQuery);
+
+     private:
+      bool validate(const model::GetAccountAsset& query);
+      bool validate(const model::GetAccount& query);
+      bool validate(const model::GetSignatories& query);
+      bool validate(const model::GetAccountAssetTransactions& query);
+      bool validate(const model::GetAccountTransactions& query);
+
```
Comment (takemiyamakoto) [inline irohad/model/query_execution.hpp]: The following lines of code are somewhat hard to read

File: test/module/irohad/validation/query_execution.cpp [hunk 5]
```diff
@@ -0,0 +1,276 @@
...
+#include "model/queries/responses/account_response.hpp"
+#include "model/queries/responses/error_response.hpp"
+
+using ::testing::Return;
+using ::testing::AtLeast;
+using ::testing::_;
+using ::testing::AllOf;
+
+class WSVQueriesMock : public iroha::ametsuchi::WsvQuery {
+ public:
```
Comment (takemiyamakoto) [inline test/module/irohad/validation/query_execution.cpp]: This code is really hard to read

File: irohad/model/query_execution.hpp (hunk 4 above)
Comment (grimadas) [inline irohad/model/query_execution.hpp]: Thanks, I'll add comments

File: irohad/model/queries/get_account_assets.hpp (hunk 2 above)
Comment (neewy) [inline irohad/model/queries/get_account_assets.hpp]: Rename either file or struct then

**Comment by neewy** [comment]
For some reason develop stopped building after merging this commit. I am reverting changes

**Pull request merged**